### ISSUE TITLE
Fix code gen for construction of a matrix via literals

### ISSUE THREAD

**Issue by seantalts** [stan-dev/stanc3#242]
```stan
functions {
  matrix Foo() {
    return [ [1,2,3,4,5,6,7,8,9,10]
           , [1,2,3,4,5,6,7,8,9,10]
           , [1,2,3,4,5,6,7,8,9,10] ];
  }
}
```

We should add this to the mother model too.

Ref: https://discourse.mc-stan.org/t/first-stanc3-release-candidate/9692/6

**Comment by VMatthijs** [comment]
@seantalts , didn't you say this had been fixed?

**Comment by seantalts** [comment]
Yep, my bad - didn't see the issue. Fixed by #250.
